docs/standard/serialization/binaryformatter-migration-guide/choose-a-serializer.md

```diff
@@ -0,0 +1,75 @@
+| Serializing public types                       | ✔️              | ✔️                      | ✔️                    | ✔️                       | ✔️                               |
+| Serializing non-public types                   | ✔️              | ✔️                      | ✔️                    | ✔️ (resolver required)   | ✔️                               |
+| Serializing fields                             | ✔️              | ✔️ (opt in)             | ✔️                    | ✔️ (attribute required)  | ✔️ (attribute required)          |
+| Serializing non-public fields                  | ✔️              | ✔️ (resolver required)  | ✔️                    | ✔️ (resolver required)   | ✔️ (attribute required)          |
+| Serializing properties                         | ✔️<sup>*</sup>  | ✔️                      | ✔️                    | ✔️ (attribute required)  | ✔️ (attribute required)          |
+| Deserializing readonly members                 | ✔️              | ✔️ (attribute required) | ✔️                    | ✔️                       | ✔️ (parameterless ctor required) |
+| Polymorphic type hierarchy                     | ✔️              | ✔️ (attribute required) | ✔️                    | ✔️ (attribute required)  | ✔️ (attribute required)          |
+| AOT support                                    | ❌️              | ✔️                      | ❌                    | ✔️                       | ❌ (planned)                     |
+
+## JSON using System.Text.Json
+
+The [`System.Text.Json`](../system-text-json/overview.md) library is a modern serializer that emphasizes security, high performance, and low memory allocation for the JavaScript Object Notation (JSON) format. JSON is human-readable and has broad cross-platform support. While text-based format is not as compact as binary formats, it can be significantly reduced in size through compression.
+
+Serialization excludes non-public and readonly members unless specifically handled through attributes and constructors. System.Text.Json also supports [custom serialization and deserialization](../system-text-json/custom-contracts.md) for more control over how types are converted into JSON and vice versa. System.Text.Json does not support the `[Serializable]` attribute.
+
+[Migrate to System.Text.Json](./migrate-to-system-text-json.md).
+
+## XML using System.Runtime.Serialization.DataContractSerializer
+
+`DataContractSerializer` was introduced in .NET Framework 3.0 and is used to serialize and deserialize data sent in Windows Communication Foundation (WCF) messages. `DataContractSerializer` is an XML serializer that **fully supports the serialization programming model that was used by the `BinaryFormatter`**. It requires the known types to be specified up-front (but most .NET collections and primitive types are on a default allow-list and don't need to be specified). It's the serializer that requires the least amount of effort to migrate to.
```

Comment:
>  It's the serializer that requires the least amount of effort to migrate to.

Why do feel that is true? I'd imagine S.T.J would be easier, partially because it's more modern, but also because it doesn't generally require annotating types.

Reply:
Ah, it's explained in the linked topic:

> Why? Because it **fully supports the serialization programming model that was used by `BinaryFormatter`**. All the types that are already marked as `[Serializable]` or implement `ISerializable` can be serialized with `DataContractSerializer`.

The second sentence is what did it for me.


```suggestion
`DataContractSerializer` was introduced in .NET Framework 3.0 and is used to serialize and deserialize data sent in Windows Communication Foundation (WCF) messages. `DataContractSerializer` is an XML serializer that **fully supports the serialization programming model that was used by the `BinaryFormatter`**, which means it honors the `[Serializable]` attribute and implementing `ISerializable`. Hence, it's the serializer that requires the least amount of effort to migrate to. It does, however, require the known types to be specified up-front (but most .NET collections and primitive types are on a default allow-list and don't need to be specified). 
```